src/main/java/dev/hephaestus/glowcase/block/entity/ItemDisplayBlockEntity.java

```diff
@@ -190,19 +191,29 @@ else return switch(this.givesItem) {
 		};
 	}
 	
-	public void giveTo(PlayerEntity player, Hand hand) {
-		player.setStackInHand(hand, getDisplayedStack().copy());
+	public void giveTo(PlayerEntity player) {
+		ItemStack itemStack = player.getStackInHand(Hand.MAIN_HAND);
+		boolean holdingSameAsDisplay = ItemStack.areItemsEqual(getDisplayedStack(), itemStack);
```

Comment:
Subtle bug: this does not check for components. If you obtained some special unique renamed version of the item in the display it'll give you another one of the special item, even if the one in the display is just an unnamed item. Using `areItemsAndComponentsEqual` would be better here

Reply:
Line 196 was copied verbatim from a different part of the code responsible for showing the item display block edit screen when holding the same item. Should that be fixed the same way?

Reply:
Nah, I think it's useful to be able to edit the block even if you don't have the exact item, since they can get pretty complex

Reply:
Fixed with 92ba0a6783717c24135712743c1854d2d536aaf7